_posts/2018-01-09-dagger2-android-subcomponents.md

```diff
@@ -0,0 +1,205 @@
+---
+layout: post
+title:  "Using Dagger2 SubComponents to Propagate Dependancies"
+date:   2018-01-09 8:00:00
+author: abhishek
+categories: Android, Dependancy Injection
+---
+Dependancy Injection makes your code scalable and testable. `Dagger2` is almost de-facto Dependancy Injection framework in Java these days, and, Its supported and used by Google which makes it first choice of all Android Developers.
+
+In this post I am going to demonstrate use of `Subcomponents` in a typical Android app and show how Dependancy Injection is done from top to bottom. [In this app](https://github.com/abhishekBansal/android-mvp-retrofit2-dagger2-rxjava2-testing) user can see list of movies and click on any list item to see its details. Movie list is fetched from remote API, result of this API are cached in local database and this database is then used to present movie details on movie detail screen.
```

Comment:
Maybe change the language to "We will take this movie app as an example and add dependency injection to it" and then add a screenshot or two?